### DISCUSSION TITLE
Loosing volume configuration after container update

### DISCUSSION THREAD

**Discussion by tiagodj** [containrrr/watchtower#799] [Q&A] [answered]
Hi,

I have a docker container started using the option "-v /dev/bus/usb:/dev/bus/usb". But when watchtower updates that container, it looses this volume.

Why?

**Comment by piksel** [comment, marked as answer]
What do you mean by "loses"? The new container have no volume at all?

Have you tried running with `--debug`? I don't know of any reason for it to not work. Perhaps it's kernel-related, since it's in /dev?

**Reply by tiagodj** [reply]
I did some digging and found out that the volumes are there if I run "docker inspect". They don't show in the UI, and that led me to believe that they are not mapped. Sorry about that.

But the problem persists. When watchtower starts the container, the application (home assistant) can't access the USB devices which are mapped using those parameters.

Only when I delete it and start it manually, that it works as expected. This is the command used to start the container manually:

```
sudo docker run \
    --init \
    --name="home-assistant" \
    --restart=always \
    --net=host \
    --privileged \
    -itd \
    -e "TZ=America/Toronto" \
    -v /volume1/docker/homeassistant/config:/config \
    -v /volume1/docker/docker.sock:/var/run/docker.sock \
    -v /dev/bus/usb:/dev/bus/usb \
    -v /var/run/dbus:/var/run/dbus \
    homeassistant/home-assistant:latest
```

Once a new version comes out I'll run watchtower with --debug to see what happens.

**Reply by piksel** [reply]
Yeah, there could be any number of issues with this setup. I would suggest to add every separate device as a `--device` instead of using the whole bus directory as a volume. But I don't think we can do much in watchtower to mitigate this problem, sorry.
